src/codeu/chat/client/ClientUser.java

```diff
@@ -109,96 +109,7 @@ public void showCurrent() {
-  public boolean addUser(String name) {
-    final boolean validInputs = isValidName(name);
-    
-    //Check there are users and if there are, get the users in a list
-      if(!usersById.isEmpty()){
-      	Iterator<User> users = usersById.values().iterator(); 
-      	while(users.hasNext()){
-      		if(users.next().name.toUpperCase().equals(name.toUpperCase())){
-      			System.out.format("Error: user %s not created due to duplicate name.\n", name);
-      			return false;           		
-      		}
-      	 }
-      	 //has gone through all names and none matched and check that the name doesn't already exist
-      	 final User user = (validInputs) ? controller.newUser(name) : null;
-	
-	
-		//check that the user being created isn't empty
-    	if (user == null) {
-     	 System.out.format("Error: user not created - %s.\n",
-          	(validInputs) ? "server failure" : "bad input value");
-        	return false; 
-        
-   	 	} else {
-      	LOG.info("New user complete, Name= \"%s\" UUID=%s", user.name, user.id);
-      	updateUsers();
-      	return true; 
-    	}  	 
-      }
-       //the users map is empty, so the user should be added and that the input is valid
-	else{
-    	final User user = (validInputs) ? controller.newUser(name) : null;
-	
-		//check that the user being created isn't empty
-    	if (user == null) {
-     	 System.out.format("Error: user not created - %s.\n",
-          	(validInputs) ? "server failure" : "bad input value");
-        return false; 
-        
-   	 	} else {
-      	LOG.info("New user complete, Name= \"%s\" UUID=%s", user.name, user.id);
-      	updateUsers();
-      	return true; 
-    	}
-    } 
-  }
-  
- 
-    public boolean addUser(String name) {
-    final boolean validInputs = isValidName(name);
-
-	if(usersById.containsValue(name)){
-		System.out.format("Error: user not created due to duplicate name.\n");
-      	return false;
-	}
-    
-    final User user = (validInputs) ? controller.newUser(name) : null;
-
-    if (user == null) {
-      System.out.format("Error: user not created - %s.\n",
-          (validInputs) ? "server failure" : "bad input value");
-          return false; 
-    } else {
-      LOG.info("New user complete, Name= \"%s\" UUID=%s", user.name, user.id);
-      updateUsers();
-      return true; 
-    }
-    //return false; 
-  } 
-  */
-  /*
-  public boolean addUser(String name) {
-    final boolean validInputs = isValidName(name);
-    
-    if(validInputs){
-    	final User user = controller.newUser(name);
-    	
-      	LOG.info("New user complete, Name= \"%s\" UUID=%s", user.name, user.id);
-      	updateUsers();
-        	
-    } else{
-      	System.out.format("Error: user not created - %s.\n",
-        (validInputs) ? "server failure" : "bad input value");
-    }
-    
-    return validInputs;
-  }
-  */
-  
+  //change the type to a boolean, so we can check if the user was created or not in the GUI
```

Comment:
Great documentation! The only thing I would say here is that you don't need the "change the type to a boolean part", because someone just reading the code wants to know what this method does, not what the code has been changed from.